scipy/interpolate/interpolate.py

```diff
@@ -36,13 +37,21 @@ def lagrange(x, w):
     ----------
     x : array_like
         `x` represents the x-coordinates of a set of datapoints.
-    w : array_like
-        `w` represents the y-coordinates of a set of datapoints, i.e., f(`x`).
+    y : array_like
+        `y` represents the y-coordinates of a set of datapoints, i.e., f(`x`).
+    reduce : bool
+        `reduce = True` represents use reduced Lagrange coefficients.
+        `reduce = False` represents use complete Lagrange coefficients (the size
+        of the coefficient array equals the size of x).
+        Default is `True`.
```

Comment:
TBH, I've no idea what reduced or complete Lagrange coefficients are. This does not seem to be a term my google-fu finds either. Best define what the switch does like you did in brackets and definitely leave the default back-compatible.